server/api/github/callback.ts

```diff
@@ -29,11 +29,13 @@ export default defineEventHandler(async (event) => {
   const createdUser = await db
     .insert(userSchema)
     .values({
+      id: user.data.id,
```

Comment:
I think it would be better if you add a new property like `forgeRemoteID` and store it into that one. Especially if we allow multiple forges later to prevent id clashing.

Reply:
@anbraten what should then be a unique id from the user info we get from github api in your opinion?

Reply:
@anbraten what do you think about using a composite key in combination of the provider name and the id from there?

Reply:
I would leave id to be our internal user-id and add a property `forgeRemoteID` and in the callback we can check if the user exists already using a get with `forgeRemoteID == user.data.id` and if not create a user otherwise update the existing one.

Reply:
Or we could create a sub-document / new relation:

```
user-connected-forges:
- user-id
- forge-id (later we can add a table forges when users are able to add own forges, for now we leave this property out)
- forge-name (simply `github` or `gitlab` until we add forge-id)
- forger-user-id (the id of the user in github or gitlab)
```

This way we could also easily allow the user to connect new forges to his existing account which would simply add a new entry to this table.

Reply:
@anbraten the idea of the table is great. Still, the issue persists with the user information being duplicated if we don't somehow create a unique id by ourselves (currently it's just auto-incrementing). Since fetching the data before (we need to query for `forgeRemoteID == user.data.id`) will add a bigger overhead I would suggest we create a unique id based on the forge id and the forge-name and let drizzle take care of the rest

Reply:
I think we always need to query the user as you wont know if the user exists in one of the forges as it can be connected to multiple ones.